core/src/main/java/org/elasticsearch/index/query/MoreLikeThisQueryBuilder.java

```diff
@@ -147,7 +204,8 @@ public Item doc(BytesReference doc) {
          * Sets to a given artificial document, that is a document that is not present in the index.
          */
         public Item doc(XContentBuilder doc) {
-            return this.doc(doc.bytes());
+            this.doc = doc != null ? doc.bytes() : null;
```

Comment:
There is a constructor for doc. However, we do have this method for bwc, so maybe we should keep it?

Concerning null index, or type, this is so that these could be initialized by the search context, and we only have this context when the query is actually being created.